Swapsea/swapsea
# Use id as foreign key, the rails way

Apologies for the mega-merge

Testing completed:
- automated specs (see README)
- Manual end to end test, including all admin functions, ICS download, swapping & emailing.

To review:
- Check the commit history first, as it's self-explanatory (There are migrations) 
- Pull & Spin it up locally

If the build passes, let's get it into Staging and see.
